grails-app/services/org/pih/warehouse/data/DataService.groovy

```diff
@@ -803,9 +803,21 @@ class DataService {
 
     def transformObject(Object object, Map includeFields) {
         Map properties = [:]
+        def value = null
         includeFields.each { fieldName, property ->
-            def value = property.tokenize('.').inject(object) { v, k -> v?."$k" }
-            properties[fieldName] = value ?: ""
+            if (property instanceof LinkedHashMap) {
+                value = property.value.tokenize('.').inject(object) { v, k -> v?."$k" }
+                if (property.format && value) {
+                    value = value.format(property.format)
```

Comment:
I assume this assumes value is a Date object. Should we formalize that? What would happen if value was a String?

Reply:
@jmiranda I couldn't find a place where date is a string so I don't think that we should formalize that. We should use format only for date formatting.

Reply:
Sorry, by "formalize" I meant that we should handle each case separately. This code works because you only use the format attribute for dates, but someone could set value to a string or any other object, set the format attribute, and then this would fail. 

So use instanceof or some other means to formally (explicitly) handle these cases. Since string objects don't have a format method (note that String.class does have a static format method) we should probably throw an exception.

Reply:
@jmiranda Okay, sure I will.

Reply:
Actually, here's one thing we can do for now just so we don't get bogged down. Change "format" to "dateFormat", then we can assume the object is a date.

Reply:
@jmiranda Done.